adapter/internal/oasparser/model/open_api.go

```diff
@@ -198,8 +205,25 @@ func getOperationLevelDetails(operation *openapi3.Operation, method string) *Ope
 		securityArray[i] = security
 	}
 	logger.LoggerOasparser.Debugf("Security array %v", securityArray)
-	return NewOperation(method, securityArray, extensions)
+	return NewOperation(method, securityArray, extensions, mockedAPIConfig)
+
+}
 
+// getMockedApiConfig recieves xMediationScriptValue, mockedApiConfig pointer value and method name. It unmrashalls the xMediationScript string 
+// to mockedApiConfig struct type.
+func getMockedAPIConfig(xMediationScriptValue interface{}, mockedAPIConfig *MockedAPIConfig, method string) {
+	if str, ok := xMediationScriptValue.(string); ok {
+		isValidJSONString := json.Valid([]byte(str))
+		if isValidJSONString {
+			unmarshalError := json.Unmarshal([]byte(str), &mockedAPIConfig)
+			if unmarshalError != nil {
+				logger.LoggerOasparser.Errorf("Error while unmarshalling JSON for method %v. Error: %v", method, unmarshalError)
+			} 
+		} else {
+			logger.LoggerOasparser.Errorf("Invalid JSON value received for mocked API implementation's %v operation.", method)
+		}
+	}
+	logger.LoggerOasparser.Debugf("x-mediation-script value processed successfully for the %v operation.", method)
```

Comment:
if `unmarshalError` is an error, it will log line 220, and also logs 226 (success)

Reply:
Looks like we need to return an error after each error log